controllers/artefacts.rb

```diff
@@ -57,6 +57,111 @@ class ArtefactsController < ApplicationController
+                    reply handle_resource_request(LinkedData::Models::Class, page, size, attributes)
+                else
+                    reply empty_page(page, size)
+                end
+            end
+          
+            get '/properties' do
+                ont, latest_submission = get_latest_submission
+                begin
+                    props = ont.properties(latest_submission)
+                rescue LinkedData::Models::Ontology::ParsedSubmissionError => e
+                    error 404, e.message
+                end
+                
+                _, page, size = settings_params(LinkedData::Models::OntologyProperty).first(3)
+                reply Goo::Base::Page.new(page, size, props.length, props)
+            end
+          
+            %w[individuals schemes collections labels].each do |resource_type|
+                get "/#{resource_type}" do
+                    model_class = case resource_type
+                        when 'individuals' then LinkedData::Models::Instance
+                        when 'schemes' then LinkedData::Models::SKOS::Scheme
+                        when 'collections' then LinkedData::Models::SKOS::Collection
+                        when 'labels' then LinkedData::Models::SKOS::Label
+                    end
+                    
+                    attributes, page, size = settings_params(model_class).first(3)
+                    attributes.reject! { |x| x.is_a?(Hash) }
+                    reply handle_resource_request(model_class, page, size, attributes)
+                end
+            end
+          
+            private
+            
+            def empty_page(page, size)
+                Goo::Base::Page.new(page, size, 0, [])
+            end
+            
+            def handle_resource_request(model, page, size, attributes)
+                _, latest_submission = get_latest_submission
+                check_last_modified_segment(model, [@params["artefactID"]])
+                model.where.in(latest_submission).include(attributes).page(page, size).all
+            end
+          
+            def get_latest_submission
+                @ontology ||= Ontology.find(@params["artefactID"]).first
+                error 404, "You must provide a valid `artefactID` to retrieve an artefact" if @ontology.nil?
+                
+                check_last_modified(@ontology)
+                
+                @latest_submission ||= @ontology.latest_submission(status: [:RDF])
+                error 404, "Artefact #{@params["artefactID"]} distribution not found." if @latest_submission.nil?
+                
+                unless @latest_submission.ready?(status: [:RDF])
+                    error 404, "Artefact #{params["artefactID"]} distribution #{@latest_submission.submissionId} has not been parsed."
+                end
+                
+                @latest_submission.bring(*LinkedData::Models::OntologySubmission.goo_attrs_to_load([]))
```

Comment:
you don't need to load everything here.

Reply:
yes we not only `ontology: [:acronym]` to make it work